src/lib/purgeBigNumber.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Interface for objects with a toNumber method (like BigNumber)
+ */
+interface BigNumberLike {
+  toNumber: () => number;
+}
+
+/**
+ * Type guard to check if an object is BigNumber-like
+ */
+function isBigNumberLike(obj: unknown): obj is BigNumberLike {
+  return (
+    obj !== null &&
+    typeof obj === 'object' &&
+    'toNumber' in obj &&
+    typeof (obj as any).toNumber === 'function'
+  );
+}
+
+/**
+ * Recursive type that converts BigNumberLike to number at all levels
+ */
+type ConvertBigNumberToNumber<T> = T extends BigNumberLike
+  ? number
+  : T extends Array<infer U>
+    ? Array<ConvertBigNumberToNumber<U>>
+    : T extends object
+      ? { [K in keyof T]: ConvertBigNumberToNumber<T[K]> }
+      : T;
+
+/**
+ * Recursively converts BigNumber objects to numbers in an object structure
+ * @param obj - The value to process (object, array, or primitive)
+ * @returns A new value with all BigNumber instances converted to numbers
+ */
+export function purgeBigNumbers<T>(obj: T): ConvertBigNumberToNumber<T> {
```

Comment:
why do the returns need to be casted to any? shouldn't they be `number | undefined | null`?